packages/contracts/docs/osx/01-how-it-works/02-framework/02-plugin-management/02-plugin-setup/index.md

```diff
@@ -1,72 +1,78 @@
 
-1. A DAO builder selects a plugin with a specific version from the UI to install, uninstall, or update. Depending on the case, the `prepareInstallation`, `prepareUpdate`, or `prepareUninstallation` method in the `PluginSetup` contract associated with that version is called through the `PluginSetupProcessor` (and creates a unique setup ID).
-2. The `PluginSetup` contract deploys all the contracts and gathers addresses and other input arguments required for the plugin setup. This can include:
+2. The `PluginSetup` contract deploys all the contracts and gathers addresses and other input arguments required for the installation/uninstallation/upgrade instructions. This can include:
 
    - deployment of new contracts
    - initialization of new storage variables
    - deprecating/decomissioning outdated (helper) contracts
+   - governance settings or other attributes
    - ...
 
-   Because the addresses of all associated contracts are now known, a static permission list can be emitted, hashed, and stored on-chain
+   Because the addresses of all associated contracts are now known, a static permission list can be emitted, hashed, and stored on-chain.
 
-3. The list containing the required permissions is then proposed as an `Action[]` array for processing in a proposal through a governance plugin of the installing DAO.
+3. Once the Plugin has been prepared for installation, this encoded action is added to the `Action[]` array to be executed when a proposal passes. For a plugin to be installed, it needs to be approved by the governance mechanism of the organization.
 
 :::info
-The governance plugin can be a simple majority vote, an optimistic process or an admin governance plugin that does not involve a waiting period. It can be any governance mechanism existing within the DAO.
+The governance plugin can be a simple majority vote, an optimistic process or an admin governance plugin that does not involve a waiting period. It can be any governance mechanism existing within the DAO which has access to the DAO's `execute` permission.
 :::
 
-This gives the DAO time to see and check which permissions the `PluginSetup` contract request before processing them.
+This gives the DAO members the opportunity to check which permissions the `PluginSetup` contract request before granting/revoking them.
+
 Plugin setup proposals must be carefully examined as they can be a potential security risk if the `PluginSetup` contract comes from an untrusted source. To learn more visit the [Security](./01-security-risk-mitigation.md) section.
 
 <!-- TODO: add a costs sections
 
 Optionally, the proposer can also request refunds for the gas spent for the preparation of the plugin in the proposal.
 -->
 
-### Setup Application
+### Applying the action
+
+After this initial preparation transaction, all contracts and addresses related to the plugin, as well as their permissions, are known and the DAO can decide if the proposal should be accepted or denied.
```

Comment:
The DAO doesn't decide it, it's the members of an existing governance plugin who decide about the proposal.
```suggestion
After this initial preparation transaction, the addresses and permissions related to the plugin become apparent. The members of a governance plugin with permissions can decide if the installation proposal should be accepted or denied.
```